### PR TITLE
[Modular] Xenoarchaeology

### PR DESCRIPTION
**Pull request by jjpark-kb**
## About The Pull Request

This PR adds a Science Research Outpost on the lower level of Lavaland. In this Outpost, there is Xenoarchaeology, which is a new branch of science geared towards mining and recovering artifacts from strange rocks. Another new department is Xenobotany, which requires some progress in Xenoarchaeology. Get some seeds from flora fossils and get started on that. On the outpost, there are three dorm rooms, an engine and atmospherics room, and four empty 'maint' rooms. 

How do you get to the outpost? Take public mining shuttle. It has been moved to the lower level of lavaland. You could also take the elevator!

To increase RP and perhaps allow trading to occur, there are also roundstart permanent lifebringers and free golems. They are neutral in terms of factions, compared to Ash walkers, who are somewhat aggressive. If ashwalkers are allowed to be roundstart, why not golems and podpeople.

## Why It's Good For The Game

More content for science and makes life a little easy and more enjoyable for certain departments. Roundstart lifebringers and free golems will allow for more RP regarding trading, socialization, etc.

## Changelog
:cl:
add: Added Xenoarchaeology
add: Added Xenobotany
add: Public mining goes to the lower lavaland
add: Added roundstart Lifebringers and Free Golems
add: Added public mining to Pubbystation
add: Xenoarch Elevators to Engineering and Science.
tweak: Changed Scientists from 3/5 to 7/9 (spawn/total) positions.
/:cl:

### CODE REVIEW

File: modular_skyrat/code/modules/research/xenoarch/strange_rock.dm [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+			itemactualdepth = rand(itembasedepth - itemsafedepth,itembasedepth)
+		if(200)
+			chosenitem = pickweight(GLOB.ult_artifact)
+			itembasedepth = rand(70,100)
+			itemsafedepth = rand(12,14)
+			itemactualdepth = rand(itembasedepth - itemsafedepth,itembasedepth)
+	..()
+
+/obj/item/strangerock/attackby(obj/item/W, mob/user, params)
+	if(istype(W,/obj/item/xenoarch/clean/hammer/cm1))
```
Comment (NotRanged) [inline modular_skyrat/code/modules/research/xenoarch/strange_rock.dm:38]: Line 38 all the way to line 157, all 120 of them, could be condensed into only 26 lines.

```suggestion
	if(istype(W, /obj/item/xenoarch/clean/hammer/))
		if(tryagain)
			to_chat(user,"You are already mining this.")
			return
		tryagain = TRUE
		var/obj/item/xenoarch/clean/hammer/HM = W
		playsound(loc, HM.usesound, 50, 1, -1)
		if(!do_after(user,HM.cleandepth * 4,target = src))
			to_chat(user,"You must stand still to clean.")
			tryagain = FALSE
			return
		dugdepth += HM.cleandepth
		if(dugdepth > itemactualdepth)
			to_chat(user,"The strange rock crumbles, destroying anything that could have been recovered.")
			qdel(src)
			return
		tryagain = FALSE
```

It's all duplicate code. You could even add 'cleanspeed' or something as a variable to the hammers themselves and use that as a modifier in the do_after if you want to keep that.

File: modular_skyrat/code/modules/research/xenoarch/strange_rock.dm [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+			to_chat(user,"You must stand still to clean.")
+			tryagain = FALSE
+			return
+		dugdepth += HM.cleandepth
+		if(dugdepth > itemactualdepth)
+			to_chat(user,"The strange rock crumbles, destroying anything that could have been recovered.")
+			qdel(src)
+			return
+		tryagain = FALSE
+	if(istype(W,/obj/item/xenoarch/clean/brush))
```
Comment (NotRanged) [inline modular_skyrat/code/modules/research/xenoarch/strange_rock.dm:174]: Same for the brushes. Only difference is cleaning speed(?), could make that a variable of item/xenoarch/clean and save dozens of lines of duplicate code

File: modular_skyrat/code/modules/research/xenoarch/strange_rock.dm [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+			chosenitem = pickweight(GLOB.bas_artifact)
+			itembasedepth = rand(30,40)
+			itemsafedepth = rand(3,6)
+			itemactualdepth = rand(itembasedepth - itemsafedepth,itembasedepth)
+		if(160 to 199)
+			chosenitem = pickweight(GLOB.adv_artifact)
+			itembasedepth = rand(40,60)
+			itemsafedepth = rand(6,12)
+			itemactualdepth = rand(itembasedepth - itemsafedepth,itembasedepth)
+		if(200)
```
Comment (NotRanged) [inline modular_skyrat/code/modules/research/xenoarch/strange_rock.dm:30]: 1 in 200 is reaaaallly rare, are you sure about this? That's generally rarer than getting a *super epic legendary drop* from shitty lootbox mobile games.

File: modular_skyrat/code/modules/research/xenoarch/xenobotany/grown/vaporsac.dm [hunk 4]
```diff
@@ -0,0 +1,28 @@
+/obj/item/seeds/vaporsac
```
Comment (NotRanged) [inline modular_skyrat/code/modules/research/xenoarch/xenobotany/grown/vaporsac.dm:1]: Do all these plants really need their own .dm files? I suggest keeping it all in a single .dm for simplicity and ease of use, something like xenoarchflora.dm

Comment (jjpark-kb) [reply modular_skyrat/code/modules/research/xenoarch/xenobotany/grown/vaporsac.dm:1]: The plants are actually done using this, see the hydroponics module.

Comment (NotRanged) [reply modular_skyrat/code/modules/research/xenoarch/xenobotany/grown/vaporsac.dm:1]: Ah, wasn't aware of that, my bad.

File: modular_skyrat/code/modules/research/xenoarch/strange_rock.dm (hunk 1 above)
Comment (jjpark-kb) [reply modular_skyrat/code/modules/research/xenoarch/strange_rock.dm:38]: working on this. I can condense the hammers, dont feel like condensing the brushes at the moment

File: modular_skyrat/code/modules/research/xenoarch/strange_rock.dm (hunk 2 above)
Comment (jjpark-kb) [reply modular_skyrat/code/modules/research/xenoarch/strange_rock.dm:174]: Addressed in a previous comment: Only doing hammers, not brushes atm.

File: modular_skyrat/code/modules/research/xenoarch/strange_rock.dm (hunk 3 above)
Comment (jjpark-kb) [reply modular_skyrat/code/modules/research/xenoarch/strange_rock.dm:30]: Ill lower it to 100, we can see how bad it is.

**Pull request merged**